spec/priorityQueue.js

```diff
@@ -0,0 +1,126 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import priorityQueue from '../src/priorityQueue'
+
+chai.use(chaiChange)
+
+describe('priorityQueue', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(priorityQueue).to.be.a('function')
+  })
+
+  describe('enqueue()', () => {
+    it('pushes an element to the queue.', () => {
+      const queue = new priorityQueue()
+
+      expect(() => queue.enqueue('foo', '1'))
+        .to.alter(() => queue.length(), { from: 0, to: 1 })
+    })
+
+    it('pushes an element to the back of the queue.', () => {
+      const queue = new priorityQueue()
+      queue.enqueue('foo', 1)
+
+      expect(queue.back())
+        .to.eql({entry: 'foo', priority: 1})
+    })
+  })
+
+  describe('dequeue()', ()=> {
+    it('returns and removes the front element in the queue.', () => {
+      const queue = new priorityQueue()
+      queue.enqueue('foo')
+
+      expect(() => queue.dequeue())
+        .to.alter(() => queue.length(), {from: 1, to: 0 })
+    })
+
+    it('returns null if the queue is empty.', () => {
+      const queue = new priorityQueue()
+
+      expect(queue.dequeue())
+        .to.be.null
+    })
+    it('removes returned element from queue', () => {
+      const queue = new priorityQueue()
+
+      expect(queue.dequeue())
+        .to.be.null
+    })
```

Comment:
Should also test if you're returning the element in the back of the queue when you dequeue it (spec: **returns** and removes the front element in the queue). 

Right now, I just see that you test for the length of the resulting queue, not the value returned by `dequeue()`.